File: video_router/adapters/hailuo23.py
"""
hailuo23.py — Hailuo 2.3 adaptoru (HAZIR — model 23217)
=======================================================
Hailuo pipeline: start_only sahneler buraya gelir.
Ortak akis _hailuo.submit (2.0 ile ayni; TEK fark model ID).

Model ID: 23217 (2.0 = 23210). Diger parametreler 2.0 ile ayni: resolution 1080,
duration 6, projectID klasorden. Token/cookie/project: VIDEO klasorunde.

referenceMode notu: hl_generate_video start_only'de "start-frame" + tek dosya (frameType 0)
gonderir. Curl'de "start-end-frames" + tek dosya goruldu (panel state'i olabilir).
ONCE "start-frame" ile denenir; 2.3 reddederse curl'deki start-end-frames taklit edilecek.
Poll 2.0 ile ayni (vid_id bazli my-work-detail, proje bagimsiz, watermarksiz).
"""

from .. import core
from . import _hailuo

MODEL_ID = "23217"


def _generate(job: core.Job):
    return _hailuo.submit(job, model_id=MODEL_ID, tag="Hailuo 2.3")


SPEC = core.register(core.AdapterSpec(
    key="hailuo2.3",
    provider="hailuo",
    model_tag="hailuo23",
    modes={"start_only"},
    ready=True,
    generate=_generate,
    token_files=["hailuo_token.txt", "hailuo_cookie.txt", "hailuo_project.txt"],  # VIDEO klasorunde
    description="Hailuo 2.3 (start_only, model 23217) — HAZIR.",
))

File: src/lib/pipeline/router/adapters/hailuo23.ts
import { submitHailuoJob } from "../../hailuo";
import { HAILUO_MODEL_23 } from "../../hailuo/constants";
import { type Job, register } from "../core";

register({
  key: "hailuo2.3",
  provider: "hailuo",
  modelTag: "hailuo23",
  modes: new Set(["start_only"]),
  ready: true,
  description: "Hailuo 2.3 (start_only, model 23217)",
  generate: async (job: Job) => {
    return submitHailuoJob({
      videoDir: job.videoDir,
      modelId: HAILUO_MODEL_23,
      label: String(job.scene.label || ""),
      frameMode: String(job.scene.frame_mode || "start_only"),
      prompt: job.prompt,
      startImage: job.startImage,
      endImage: job.endImage,
      outPath: job.outPath,
      resumeVidId: job.resumeVidId,
      duration: job.duration,
      resolution: job.resolution ?? undefined,
      promptOptimizer: job.promptOptimizer !== false,
      skipQueueGate: job.skipQueueGate,
      preGenerate: job.preGenerate,
      onSubmit: job.onSubmit,
    });
  },
});
